components/org.wso2.carbon.event.simulator.core/src/main/java/org/wso2/carbon/event/simulator/core/internal/generator/random/util/PropertyBasedAttrGenerator.java

```diff
@@ -69,16 +71,21 @@ public PropertyBasedAttrGenerator() {
      * @throws InvalidConfigException if the attribute configuration contains an invalid propertyType
      */
     @Override
-    public void validateAttributeConfiguration(JSONObject attributeConfig) throws InvalidConfigException {
+    public void validateAttributeConfiguration(Attribute.Type attributeType, JSONObject attributeConfig)
+            throws InvalidConfigException {
         if (checkAvailability(attributeConfig, EventSimulatorConstants.PROPERTY_BASED_ATTRIBUTE_PROPERTY)) {
+            String propertyType = attributeConfig
+                    .getString(EventSimulatorConstants.PROPERTY_BASED_ATTRIBUTE_PROPERTY);
             try {
-                PropertyType.valueOf(attributeConfig
-                        .getString(EventSimulatorConstants.PROPERTY_BASED_ATTRIBUTE_PROPERTY));
-            } catch (IllegalArgumentException e) {
-                throw new EventGenerationException("Invalid property type '" + propertyBasedAttrConfig.getProperty() +
-                        "' provided for random event generation using " +
-                        RandomAttributeGenerator.RandomDataGeneratorType.PROPERTY_BASED + " attribute generator. " +
-                        "Invalid attribute configuration provided : " + attributeConfig.toString());
+                PropertyType.valueOf(propertyType);
+                propertyBasedAttrConfig.setProperty(PropertyType.valueOf(propertyType));
+                for (int i = 0; i < 5; i++) {
+                    DataParser.parse(attributeType, generateAttribute());
```

Comment:
Since the data value is randomly generated,I check for 5 times to reduce the probability of accepting an invalid configuration. Shall i remove it and just check once?